tests/core/fsck-on-boot/task.yaml

```diff
@@ -54,7 +71,11 @@ execute: |
       # Use offset 65 as FAT32 kicks in for devices larger than 32MB
       printf "\x01" > one
       tests.cleanup defer rm -f one
-      dd if=one of=/dev/sda2 seek=65 bs=1 count=1 conv=notrunc
+      if os.query is-core16 || os.query is-core18; then
+        dd if=/dev/disk/by-label/system-boot of=dirty skip=65 bs=1 count=1 conv=notrunc
```

Comment:
minor nitpick, but instead of duplicating the whole dd command, you could instead just have the `if` set an environment variable for the label like this:

```bash
if os.query is-core16 || os.query is-core18; then
  LABEL=system-boot
elif os.query is-core20; then
  LABEL=ubuntu-seed
else
  echo "unknown core system, please update test"
  exit 1
fi
dd if="/dev/disk/by-label/$LABEL" of=dirty skip=65 bs=1 count=1 conv=notrunc
```

Reply:
I guess we can skip the `else` branch given that the check for uc22 et al is also done above too

Reply:
/me notes that this also would be a good use for https://github.com/snapcore/spread/issues/83 if that ever becomes a thing

Reply:
totally agree, should be great to be able to define vars by system